Kudu.Core/Permissions/PermissionHandler.cs

```diff
@@ -0,0 +1,34 @@
+﻿using System.IO;
+using Kudu.Contracts.Permissions;
+using Kudu.Contracts.Settings;
+using Kudu.Core.Deployment;
+using Kudu.Core.Deployment.Generator;
+using Kudu.Core.Infrastructure;
+
+namespace Kudu.Core.Permissions
+{
+    /// <summary>
+    /// This handler is meant for Linux enviroment only
```

Comment:
I don't see how it could do that. All the classes that take in an `IPermissionHandler` will receive an instance of this class even if they never call `Chmod`.

Generally, I feel there is way too much added Ninject and parameter passing everywhere just to call Chmod is a couple places. Can't Chmod just be a static helper called conditionally, without affecting anything else?